assets/js/components/Dashboard/NewPaMessage/StationsAndZones/SelectStationsAndZones.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useState } from "react";
+import SelectStationsPage from "./SelectStationsPage";
+import SelectZonesPage from "./SelectZonesPage";
+import type { Place } from "Models/place";
+import { Page } from "../types";
+import { useScreenplayDispatchContext } from "Hooks/useScreenplayContext";
+
+interface Props {
+  places: Place[];
+  value: string[];
+  onChange: (signIds: string[]) => void;
+  navigateTo: (page: Page) => void;
+  page: Page;
+  busRoutes: string[];
+  onError: (message: string) => void;
+}
+
+const SelectStationsAndZones = ({
+  places,
+  value,
+  onChange,
+  navigateTo,
+  page,
+  busRoutes,
+  onError,
+}: Props) => {
+  const dispatch = useScreenplayDispatchContext();
+  const [signIds, setSignIds] = useState(value);
+
+  useEffect(() => {
+    dispatch({ type: "SHOW_SIDEBAR", showSidebar: false });
+
+    return () => dispatch({ type: "SHOW_SIDEBAR", showSidebar: true });
+  }, []);
```

Comment:
**question:** this is kind of odd to me. is this because the select stations and zones stuff is inside a modal now? what's the reason for the modal? (heads up: i have a bias against modals 🙂)

Reply:
No, they actually aren't in modals anymore. The problem is the conditional rendering of the Sidebar involves looking at the current route. Since both pages rendered here do not have a unique route, we can't hide the Sidebar in the same way we do everywhere else. This was kind of a "best way to make the conditional work without changing the structure of the app" way of tackling the problem. Very much open to hearing other suggestions.